### PR TITLE
Add Doxygen comments to all public `ElastixImageFilter` member functions

### PR DESCRIPTION
**Pull request by N-Dekker**
Part of issue https://github.com/SimpleITK/SimpleITK/issues/1665 "Update ElastixImageFilter and TransformixImageFilter Doxygen comments"

### CODE REVIEW

**Review by blowekamp — APPROVED** [review]
Looks good!

**Review by zivy — CHANGES_REQUESTED** [review]
Generally looks good. 

One thing that I am missing, how are the masks matched to the images, fixed or moving? For example if I have multiple moving images [MI1, MI2, MI3] and I want to add masks only to some of them [M1, none, M2] is that possible? Please add documentation explaining the constraints and expectations with respect to the information that should match between the various data containers (fixed_images, moving_images, fixed_masks, moving_masks). 

I suspect the following are expected, but not sure:
1.  size(fixed_images) == size(moving_images)
2. size(fixed_images) == size(fixed_masks), size(moving_images) == size(moving_masks).

**Comment by blowekamp** [comment]
@N-Dekker Are you able to address Ziv's comments?

**Comment by N-Dekker** [comment]
> Are you able to address Ziv's comments?

@blowekamp @zivy I'm still studying them. `ElastixImageFilterImpl::Execute` appears to check those constraints, starting here:

https://github.com/SimpleITK/SimpleITK/blob/ac3afb935291dc7830f370cb89ba54dea8e6d525/Code/ElastixTransformixWrappers/src/sitkElastixImageFilterImpl.cxx#L55-L59 

But I would need to check them myself, in order to document them properly.

**Comment by N-Dekker** [comment]
@zivy Thansk for your comment. Just added a little note about those constraints to the documentation, with help from Marius (@mstaring). Hope it's sufficient for now, even if it may not yet be complete. The aim for this PR is just to add a _short_ description to each member function.

**Comment by zivy** [comment]
Hello @N-Dekker, as a first pass on documentation this looks good. I'm sure there is more to be done, as always.

**Review by zivy — APPROVED** [review]

File: Code/ElastixTransformixWrappers/include/sitkElastixImageFilter.h [hunk 1]
```diff
@@ -55,121 +58,264 @@ class SITKElastix_EXPORT ElastixImageFilter
...
   std::map< std::string, std::vector< std::string > > ReadParameterFile( const std::string filename );
+
+  /** \brief Writes a parameter map to the file, specified by \p filename. */
   SITK_RETURN_SELF_TYPE_HEADER WriteParameterFile( const std::map< std::string, std::vector< std::string > > parameterMap, const std::string filename );
 
+  /** \brief Executes the registration, and returns the result image.
+   * \note Before executing, the number of moving images must equal the number of fixed images,
+   * the number of fixed masks must be either zero, or one, or equal to the number of fixed images,
+   * and the number of moving masks must be either zero, or one, or equal to the number of moving images
+   * /
```
Comment (blowekamp) [inline Code/ElastixTransformixWrappers/include/sitkElastixImageFilter.h:300]: The '* /' with the space seems to be causing a compilation error.
../../Code/ElastixTransformixWrappers/include/sitkElastixImageFilter.h:303:3: warning: "/*" within comment [-Wcomment]
[18351](https://github.com/SimpleITK/SimpleITK/actions/runs/3128399687/jobs/5076187945#step:6:18352)
  303 |   /** \brief Returns all transform parameter maps. */
[18352](https://github.com/SimpleITK/SimpleITK/actions/runs/3128399687/jobs/5076187945#step:6:18353)
      |    
[18353](https://github.com/SimpleITK/SimpleITK/actions/runs/3128399687/jobs/5076187945#step:6:18354)
../../Code/ElastixTransformixWrappers/src/sitkElastixImageFilter.cxx:621:1: error: no declaration matches ‘itk::simple::Image itk::simple::ElastixImageFilter::Execute()’
[18354](https://github.com/SimpleITK/SimpleITK/actions/runs/3128399687/jobs/5076187945#step:6:18355)
  621 | ElastixImageFilter
[18355](https://github.com/SimpleITK/SimpleITK/actions/runs/3128399687/jobs/5076187945#step:6:18356)
      | ^~~~~~~~~~~~~~~~~~

**Pull request merged**